JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesSparql.java

```diff
@@ -143,30 +180,120 @@ public boolean checkTimeUnitExists(String tsIRI) {
+	 * @param unit Required for Average Time Series. Temporal unit type of the averaging period for Average TimeSeries. (optional)
+	 *             Allowed values of type ChronoUnit:
+	 *             ChronoUnit.SECONDS, ChronoUnit.MINUTES, ChronoUnit.HOURS, ChronoUit.DAYS, ChronoUnit.WEEKS, ChronoUnit.MONTHS, ChronoUnit.YEARS
+	 *
+	 */
+
+	protected void initTS(String timeSeriesIRI, List<String> dataIRI, String dbURL, String timeUnit, String type, Duration duration, ChronoUnit unit) {
+		// Construct time series IRI
+		Iri tsIRI;
 		// Check whether timeseriesIRI follows IRI naming convention of namespace & local_name
-    	// If only local name is provided, iri() function would add filepath as default prefix
-    	// Every legal (full) IRI contains at least one ':' character to separate the scheme from the rest of the IRI
+		// If only local name is provided, iri() function would add filepath as default prefix
+		// Every legal (full) IRI contains at least one ':' character to separate the scheme from the rest of the IRI
 		if (Pattern.compile("\\w+\\S+:\\S+\\w+").matcher(timeSeriesIRI).matches()) {
 			tsIRI = iri(timeSeriesIRI);
 		} else {
 			throw new JPSRuntimeException(exceptionPrefix + "Time series IRI does not have valid IRI format");
 		}
 
-		// Check that the time series IRI is not yet in the Knowledge Graph
-		if (checkTimeSeriesExists(timeSeriesIRI)) {
-			throw new JPSRuntimeException(exceptionPrefix + "Time series " + timeSeriesIRI + " already in the Knowledge Graph");
+		ModifyQuery modify = Queries.MODIFY();
+		// set prefix declarations
+		modify.prefix(prefix_ontology, prefix_kb, prefix_time);
+
+		if (type.equals(null)){
+			// Check that the time series IRI is not yet in the Knowledge Graph
+			if (checkTimeSeriesExists(timeSeriesIRI)) {
+				throw new JPSRuntimeException(exceptionPrefix + "Time series " + timeSeriesIRI + " already in the Knowledge Graph");
+			}
+			// define type
+			modify.insert(tsIRI.isA(TimeSeries));
+		}
+		else if(type.equals(StepwiseCumulative)){
+			if (checkStepwiseCumulativeTimeSeriesExists(timeSeriesIRI)) {
+				throw new JPSRuntimeException(exceptionPrefix + "Stepwise Cumulative Time series " + timeSeriesIRI + " already in the Knowledge Graph");
+			}
+			// define type
+			modify.insert(tsIRI.isA(StepwiseCumulativeTimeSeries));
+		}
+		else if(type.equals(CumulativeTotal)){
+			if (checkCumulativeTotalTimeSeriesExists(timeSeriesIRI)) {
+				throw new JPSRuntimeException(exceptionPrefix + "Cumulative Total Time series " + timeSeriesIRI + " already in the Knowledge Graph");
+			}
+			// define type
+			modify.insert(tsIRI.isA(CumulativeTotalTimeSeries));
+		}
+		else if(type.equals(Instantaneous)){
+			if (checkInstantaneousTimeSeriesExists(timeSeriesIRI)) {
+				throw new JPSRuntimeException(exceptionPrefix + "Instantaneous Time series " + timeSeriesIRI + " already in the Knowledge Graph");
```

Comment:
Similar comment to a previous 1, have separate methods in the TimeSeriesClient, and pass the IRI here